### PR TITLE
Add services for Teslemetry

### PR DESCRIPTION
**Pull request by Bre77**
## Proposed change

Adds documentation for custom services

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/119119
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Introduced new custom services in the Teslemetry integration for enhanced interaction with the Tesla Fleet API:
    - Navigation to coordinates.
    - Scheduled charging and departure settings.
    - Valet mode activation.
    - Speed limit management.
    - Time of use configuration.

### CODE REVIEW

File: source/_integrations/teslemetry.markdown [hunk 1]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+
+Teslemetry provides various custom services to interact with the Tesla Fleet API directly.
+
+### Navigate to coordinates
+
+`teslemetry.navigation_gps_request`
+
+| Field         | Description                | Example                          |
+|---------------|----------------------------|----------------------------------|
+| device_id     | The vehicle's device_id    | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| device_id     | The vehicle's device ID    | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```

File: source/_integrations/teslemetry.markdown [hunk 2]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+| gps           | Dictionary of coordinates  |                                  |
+| gps.latitude  | Latitude in degrees        | -27.9699373                      |
+| gps.longitude | Longitude in degrees       | 153.4081865                      |
+| order         | Order for this destination | 1                                |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown:212]: What does order mean in this context? If the number is 3, for example, would this be the third stop on your route to destination X?

File: source/_integrations/teslemetry.markdown [hunk 3]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+
+### Set scheduled charging
+
+`teslemetry.set_scheduled_charging`
+
+| Field     | Description                           | Example                          |
+|-----------|---------------------------------------|----------------------------------|
+| device_id | The vehicle's device_id               | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| device_id | The vehicle's device ID              | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```

File: source/_integrations/teslemetry.markdown [hunk 4]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+| enable    | Enable or disable scheduled charging. | true                             |
+| time      | Time to start charging in HH:MM       | 6:00                             |
+
+### Set scheduled departure
+
+`teslemetry.set_scheduled_departure`
+
+| Field                           | Description                               | Example                          |
+|---------------------------------|-------------------------------------------|----------------------------------|
+| device_id                       | The vehicle's device_id                   | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| device_id                       | The vehicle's device ID                  | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```

File: source/_integrations/teslemetry.markdown [hunk 5]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+| order         | Order for this destination | 1                                |
+
+### Set scheduled charging
+
+`teslemetry.set_scheduled_charging`
+
+| Field     | Description                           | Example                          |
+|-----------|---------------------------------------|----------------------------------|
+| device_id | The vehicle's device_id               | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
+| enable    | Enable or disable scheduled charging. | true                             |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| enable    | Enable or disable scheduled charging | true                             |
```

File: source/_integrations/teslemetry.markdown [hunk 6]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+
+`teslemetry.set_scheduled_departure`
+
+| Field                           | Description                               | Example                          |
+|---------------------------------|-------------------------------------------|----------------------------------|
+| device_id                       | The vehicle's device_id                   | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
+| enable                          | Enable or disable scheduled departure     | true                             |
+| preconditioning_enabled         | Enable preconditioning                    | true                             |
+| preconditioning_weekdays_only   | Enable preconditioning on weekdays only   | false                            |
+| departure_time                  | Time to precondition by (HH:MM)           | 6:00                             |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]: I don't understand "time to precondition". Is this the time when preconditioning stops?

File: source/_integrations/teslemetry.markdown [hunk 7]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+
+### Valet Mode
+
+`teslemetry.valet_mode`
+
+| Field         | Description                  | Example                          |
+|---------------|------------------------------|----------------------------------|
+| device_id     | The vehicle's device_id      | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
+| enable        | Enable or disable valet mode | true                             |
+| pin           | 4 digit pin                  | 1234                             |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| pin           | 4-digit pin                  | 1234                             |
```

File: source/_integrations/teslemetry.markdown [hunk 8]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+
+### Speed Limit
+
+`teslemetry.speed_limit`
+
+| Field         | Description                   | Example                          |
+|---------------|-------------------------------|----------------------------------|
+| device_id     | The vehicle's device_id       | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| device_id     | The vehicle's device ID       | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```

File: source/_integrations/teslemetry.markdown [hunk 9]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+| enable        | Enable or disable speed limit | true                             |
+| pin           | 4 digit pin                   | 1234                             |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| pin           | 4-digit pin                   | 1234                             |
```

File: source/_integrations/teslemetry.markdown [hunk 10]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+
+### Time of use
+
+`teslemetry.time_of_use`
+
+| Field         | Description                  | Example                                                                                                          |
+|---------------|------------------------------|------------------------------------------------------------------------------------------------------------------|
+| device_id     | The energy site's device_id  | 0d462c0c4c0b064b1a91cdbd1ffcbd31                                                                                 |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| device_id     | The energy site's device ID  | 0d462c0c4c0b064b1a91cdbd1ffcbd31                                                                                 |
```

**Review by c0ffeeca7 — CHANGES_REQUESTED** [review]

File: source/_integrations/teslemetry.markdown (hunk 6 above)
Comment (Bre77) [reply source/_integrations/teslemetry.markdown]: It's the time you plan to depart, and preconditioning should be completed by this time. It technically won't stop preconditioning until you start driving.

File: source/_integrations/teslemetry.markdown [hunk 11]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+
+`teslemetry.set_scheduled_departure`
+
+| Field                           | Description                               | Example                          |
+|---------------------------------|-------------------------------------------|----------------------------------|
+| device_id                       | The vehicle's device ID                  | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
+| enable                          | Enable or disable scheduled departure     | true                             |
+| preconditioning_enabled         | Enable preconditioning                    | true                             |
+| preconditioning_weekdays_only   | Enable preconditioning on weekdays only   | false                            |
+| departure_time                  | Time to precondition by (HH:MM)           | 6:00                             |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| departure_time                  | Departure time and time when preconditioning stops (HH:MM)           | 6:00                             |
```
Ok, so maybe something like this?

File: source/_integrations/teslemetry.markdown [hunk 12]
```diff
@@ -194,3 +194,73 @@ These are the entities available in the Teslemetry integration. Not all entities
...
+| off_peak_charging_weekdays_only | Enable off-peak charging on weekdays only | false                            |
+| end_off_peak_time               | Time to complete charging by (HH:MM)      | 5:00                             |
+
+### Valet Mode
+
+`teslemetry.valet_mode`
+
+| Field         | Description                  | Example                          |
+|---------------|------------------------------|----------------------------------|
+| device_id     | The vehicle's device_id      | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```
Comment (c0ffeeca7) [inline source/_integrations/teslemetry.markdown]:
```suggestion
| device_id     | The vehicle's device ID      | 0d462c0c4c0b064b1a91cdbd1ffcbd31 |
```

File: source/_integrations/teslemetry.markdown (hunk 11 above)
Comment (Bre77) [reply source/_integrations/teslemetry.markdown]: The preconditioning doesn't technically stop at that time so I've simplified it to talk about the departure time.

**Comment by c0ffeeca7** [comment]
what about order, line 212?

File: source/_integrations/teslemetry.markdown (hunk 2 above)
Comment (Bre77) [reply source/_integrations/teslemetry.markdown:212]: I think so... Tesla isn't very clear about this themselves. I'm pretty sure order 1 would be current, 2 would be next, 3 the one after that and so on.

Comment (c0ffeeca7) [reply source/_integrations/teslemetry.markdown:212]: Oh, I see. Probably, better leave the original terminology then, like you did.

**Review by c0ffeeca7 — APPROVED** [review]
Thank you, @Bre77 :+1:

**Review by c0ffeeca7 — APPROVED** [review]

**Pull request merged**